src/sentry/replays/query.py

```diff
@@ -493,3 +421,176 @@ def _activity_score():
         ],
         alias="activity",
     )
+
+
+# A mapping of marshalable fields and theirs dependencies represented as query aliases.  If a
+# column is added which depends on another column, you must add it to this mapping.
+#
+# This mapping represents the minimum number of columns required to satisfy a field.  Without this
+# mapping we would need to select all of the fields in order to ensure that dependent fields do
+# not raise exceptions when their dependencies are not included.
+#
+# If a mapping is left as `[]` the query-alias will default to the field name.
+
+FIELD_QUERY_ALIAS_MAP: Dict[str, List[str]] = {
+    "id": ["replay_id"],
+    "title": ["title"],
+    "replayType": ["replayType"],
+    "projectId": ["projectId"],
+    "platform": ["platform"],
+    "environment": ["agg_environment"],
+    "releases": ["releases"],
+    "dist": ["dist"],
+    "traceIds": ["traceIds"],
+    "errorIds": ["errorIds"],
+    "startedAt": ["startedAt"],
+    "finishedAt": ["finishedAt"],
+    "duration": ["duration", "startedAt", "finishedAt"],
+    "urls": ["urls_sorted"],
+    "countErrors": ["countErrors"],
+    "countUrls": ["urls_sorted"],
+    "countSegments": ["countSegments"],
+    "isArchived": ["isArchived"],
+    "activity": ["activity", "countErrors", "urls_sorted", "agg_urls"],
+    "user": ["user_id", "user_email", "user_name", "user_ipAddress"],
+    "os": ["os_name", "os_version"],
+    "browser": ["browser_name", "browser_version"],
+    "device": ["device_name", "device_brand", "device_family", "device_model"],
+    "sdk": ["sdk_name", "sdk_version"],
+    "tags": ["tags.key", "tags.value"],
+    # Nested fields.  Useful for selecting searchable fields.
+    "user.id": ["user_id"],
+    "user.email": ["user_email"],
+    "user.name": ["user_name"],
+    "user.ipAddress": ["user_ipAddress"],
+    "os.name": ["os_name"],
+    "os.version": ["os_version"],
+    "browser.name": ["browser_name"],
+    "browser.version": ["browser_version"],
+    "device.name": ["device_name"],
+    "device.brand": ["device_brand"],
+    "device.family": ["device_family"],
+    "device.model": ["device_model"],
+    "sdk.name": ["sdk_name"],
+    "sdk.version": ["sdk_version"],
+}
+
+
+# A flat mapping of query aliases to column instances.  To maintain consistency, the key must
+# match the column's query alias.
+
+QUERY_ALIAS_COLUMN_MAP = {
```

Comment:
same as above -- can the snql statement be a parameter passed into a column definition?